plugin/storage/es/spanstore/reader_test.go

```diff
@@ -93,11 +94,19 @@ func withSpanReader(fn func(r *spanReaderTest)) {
 		client:    client,
 		logger:    logger,
 		logBuffer: logBuffer,
-		reader:    NewSpanReader(client, logger),
+		reader:    newSpanReader(client, logger),
 	}
 	fn(r)
 }
 
+func TestNewSpanReaderWithMetrics(t *testing.T) {
+	client := &mocks.Client{}
+	logger, _ := testutils.NewLogger() // logBuffer unneeded
+	metricsFactory := metrics.NewLocalFactory(0)
+	var reader spanstore.Reader = NewSpanReader(client, logger, metricsFactory)
```

Comment:
This one is checking the api conformance of the span reader with decorator, in contrast to the one below which checks the api conformance without the decorator.